### PR TITLE
JsonSchema: items array form and additionalItems

### PR DESCRIPTION
**Pull request by yufeih**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/107458/

Array form of items schema. Pending on #4904 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4908)

### CODE REVIEW

File: test/docfx.Test/lib/JsonSchemaTest.cs [hunk 1]
```diff
@@ -223,12 +216,25 @@ public void TestJsonSchemaConfirmance(string description, string schemaText, str
...
+            "['warning','invalid-value','Invalid value for '': '3'','file',1,5]")]
+
         // required validation
         [InlineData("{'required': []}", "{}", "")]
         [InlineData("{'required': ['a']}", "{'a': 1}", "")]
         [InlineData("{'required': ['a']}", "{'b': 1}",
             "['warning','missing-attribute','Missing required attribute: 'a'','file',1,1]")]
 
+        // boolean schema
+        [InlineData("true", "[]", "")]
```
Comment (LowEntropyBody) [inline test/docfx.Test/lib/JsonSchemaTest.cs]: why we add same content in different PRs?

File: src/docfx/lib/json/UnionTypeConverter.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    }
+                    break;
+            }
+
+            if (i == genericTypes.Length)
+            {
+                i = 0;
+            }
+
+            var args = new object[genericTypes.Length];
```
Comment (LowEntropyBody) [inline src/docfx/lib/json/UnionTypeConverter.cs]: Case: genericTypes.Length == 0 ?

File: src/docfx/lib/schema/JsonSchema.cs [hunk 3]
```diff
@@ -8,8 +8,12 @@
 
 namespace Microsoft.Docs.Build
 {
+    [JsonConverter(typeof(JsonSchemaConverter))]
     internal class JsonSchema
     {
+        public static readonly JsonSchema TrueSchema = new JsonSchema();
```
Comment (LowEntropyBody) [inline src/docfx/lib/schema/JsonSchema.cs]: Add comments?

File: test/docfx.Test/lib/JsonSchemaTest.cs (hunk 1 above)
Comment (yufeih) [reply test/docfx.Test/lib/JsonSchemaTest.cs]: yes, this PR is based on #4904, (has a dependency), when #4094 is merged, the diff will disappear.

File: src/docfx/lib/json/UnionTypeConverter.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/lib/json/UnionTypeConverter.cs]: yes, then we have a valid zero length array

**Pull request merged**